File: Q10.cs
using System;

namespace Q10
{
    class Program
    {
        static void Main(string[] args)
        {
            ushort[] idade = new ushort[20];
            byte vezes = 1, pessoas = 0;
            for (byte i = 0; i < idade.Length; i ++, vezes ++)
            {
                Console.Write("Digite a {0}ª idade: ", vezes);
                idade[i] = Convert.ToUInt16(ushort.Parse(Console.ReadLine()));
                if (idade[i] >= 18)
                {
                    pessoas ++;
                }
            }
            switch (pessoas)
            {
                case 0:
                    Console.WriteLine("Das 20 idades que você digitou, nenhuma delas são de pessoas maiores de idade.");
                    break;
                case 20:
                    Console.WriteLine("Das 20 idades que você digitou, todas elas são de pessoas maiores de idade.");
                    break;
                default:
                    Console.WriteLine("Das 20 idades que você digitou, {0} delas são de pessoas maiores de idade.", pessoas);
                    break;
            }
        }
    }
}

File: Q10.py
idade = list(range(20))
vezes = 1
pessoas = 0
for i in range(0, len(idade)):
    idade[i] = int(input(f"Digite a {vezes}ª idade: "))
    if idade[i] >= 18:
        pessoas += 1
    vezes += 1
match pessoas:
    case 0:
        print("Das 20 idades que você digitou, nenhuma delas são de pessoas maiores de idade.")
    case 20:
        print("Das 20 idades que você digitou, todas elas são de pessoas maiores de idade.")
    case _:
        print(f"Das 20 idades que você digitou, {pessoas} delas são de pessoas maiores de idade.")